### PR TITLE
Introduced cmake flag to control ONNX Editor build

### PR DESCRIPTION
**Pull request by mbencer**
Ticket: 50641

In the current implementation default value of `NGRAPH_ONNX_EDITOR_ENABLE` is OFF
ONNX Editor requires ONNX Importer. The is possible following flag combinations:
`NGRAPH_ONNX_IMPORT_ENABLE=OFF`, `NGRAPH_ONNX_EDITOR_ENABLE=OFF` - do not build editor
`NGRAPH_ONNX_IMPORT_ENABLE=ON`, `NGRAPH_ONNX_EDITOR_ENABLE=OFF` - do not build editor
`NGRAPH_ONNX_IMPORT_ENABLE=OFF`, `NGRAPH_ONNX_EDITOR_ENABLE=ON` - cmake error, do not build editor
`NGRAPH_ONNX_IMPORT_ENABLE=ON`, `NGRAPH_ONNX_EDITOR_ENABLE=ON` - build editor
`NGRAPH_ONNX_IMPORT_ENABLE=not_given/default`, `NGRAPH_ONNX_EDITOR_ENABLE=ON` - build only if the default value is ON

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +65,9 @@ function(build_ngraph)
     else()
         ngraph_set(NGRAPH_ONNX_IMPORT_ENABLE OFF)
     endif()
+    if(NGRAPH_ONNX_IMPORT_ENABLE)
+        ngraph_set(NGRAPH_ONNX_EDITOR_ENABLE ON)
+    endif()
```
Comment (postrational) [inline CMakeLists.txt:70]: Perhaps this should be folded into the block above?
```
    if(NOT (ANDROID OR WINDOWS_STORE OR (MSVC AND (ARM OR AARCH64)) ))
        ngraph_set(NGRAPH_ONNX_IMPORT_ENABLE ON)
        ngraph_set(NGRAPH_ONNX_EDITOR_ENABLE ON)
    else()
        ngraph_set(NGRAPH_ONNX_IMPORT_ENABLE OFF)
        ngraph_set(NGRAPH_ONNX_EDITOR_ENABLE OFF)
    endif()
```

File: ngraph/test/onnx/onnx_test_utils.in.cpp [hunk 2]
```diff
@@ -25,6 +27,7 @@ using namespace ngraph;
 static std::string s_manifest = "${MANIFEST}";
 using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
 
+#ifdef NGRAPH_ONNX_EDITOR_ENABLE
```
Comment (postrational) [inline ngraph/test/onnx/onnx_test_utils.in.cpp:30]: Let's move `ONNXModelEditor` tests to a separate file.

File: ngraph/CMakeLists.txt [hunk 3]
```diff
@@ -89,8 +89,12 @@ option(NGRAPH_UB_SANITIZER_ENABLE "Compiles and links with Undefined Behavior Sa
 option(NGRAPH_USE_PROTOBUF_LITE "Compiles and links with protobuf-lite" OFF)
 
 if (NGRAPH_ONNX_IMPORT_ENABLE)
+    option(NGRAPH_ONNX_EDITOR_ENABLE "Enable ONNX Editor" OFF)
```
Comment (postrational) [inline ngraph/CMakeLists.txt:92]: I think we can just set this to OFF by default, without the `if` condition here.

**Review by postrational — CHANGES_REQUESTED** [review]

File: ngraph/frontend/CMakeLists.txt [hunk 4]
```diff
@@ -5,5 +5,8 @@
 if (NGRAPH_ONNX_IMPORT_ENABLE)
     add_subdirectory(onnx_common)
     add_subdirectory(onnx_import)
-    add_subdirectory(onnx_editor)
 endif()
+
+if (NGRAPH_ONNX_EDITOR_ENABLE)
+    add_subdirectory(onnx_editor)
+endif()
```
Comment (tsocha) [inline ngraph/frontend/CMakeLists.txt:12]:
```suggestion
endif()

```

**Review by tsocha — APPROVED** [review]

File: ngraph/CMakeLists.txt (hunk 3 above)
Comment (mbencer) [reply ngraph/CMakeLists.txt:92]: done

File: ngraph/test/onnx/onnx_test_utils.in.cpp (hunk 2 above)
Comment (mbencer) [reply ngraph/test/onnx/onnx_test_utils.in.cpp]: These are no `ONNXModelEditor` tests, but `TestCase` class tests that use the editor to prepare test data (we test different input types combinations).
I configured cmake to build `onnx_test_utils.in.cpp` only if `NGRAPH_ONNX_EDITOR_ENABLE=ON`

File: CMakeLists.txt (hunk 1 above)
Comment (mbencer) [reply CMakeLists.txt]: As we've discussed offline, the better default state for onnx editor seems to be OFF - removed.
In order to provide CI check, `NGRAPH_ONNX_EDITOR_ENABLE` is set to `ON` in CI config files.

**Review by tomdol — APPROVED** [review]

**Review by postrational — APPROVED** [review]

**Pull request merged**